gatling-http/src/main/scala/io/gatling/http/check/HttpCheck.scala

```diff
@@ -29,7 +30,7 @@ import io.gatling.http.response.{ Response, ResponseBodyUsageStrategy }
  * @param scope the part of the response this check targets
  * @param responseBodyUsageStrategy how this check uses the response body
  */
-case class HttpCheck(wrapped: Check[Response], scope: HttpCheckScope, responseBodyUsageStrategy: Option[ResponseBodyUsageStrategy])
+case class HttpCheck(wrapped: Check[Response], scope: Set[HttpCheckScope], responseBodyUsageStrategy: Set[ResponseBodyUsageStrategy])
```

Comment:
I want to conduct some experiments this week on the Checks design. Let me ping you once I'm done.